bmaptools/TransRead.py

```diff
@@ -55,7 +56,44 @@
 # A list of supported compression types
 SUPPORTED_COMPRESSION_TYPES = ('bz2', 'gz', 'xz', 'lzo', 'lz4', 'tar.gz',
                                'tar.bz2', 'tar.xz', 'tar.lzo', 'tar.lz4',
-                               'zip')
+                               'zip', 'tar')
+
+# Allow some file extensions to be recognised as aliases of others
+EXTENSION_ALIASES = {
+    '.tbz2': '.tar.bz2',
+    '.tbz':  '.tar.bz2',
+    '.tb2':  '.tar.bz2',
+    '.tgz':  '.tar.gz',
+    '.txz':  '.tar.xz',
+    '.tzo':  '.tar.lzo',
+    '.tlz4': '.tar.lz4',
+    '.gzip': '.gz',
+}
+
+# These need to be checked *before* those in EXTENSION_ALIASES
+NESTED_EXTENSION_ALIASES = {
+    '.tar.gzip': '.tar.gz',
+}
+
+# Info on how to decompress different archive formats
+DECOMPRESSER_EXTENSIONS = {
+    '.bz2': {'type': "bzip2", 'program': "bzip2",  'args': "-d -c"},
+    '.gz':  {'type': "gzip",  'program': "gzip",   'args': "-d -c"},
+    '.xz':  {'type': "xz",    'program': "xz",     'args': "-d -c"},
+    '.lzo': {'type': "lzo",   'program': "lzop",   'args': "-d -c"},
+    '.lz4': {'type': "lz4",   'program': "lz4",    'args': "-d -c"},
+    '.zip': {'type': "zip",   'program': "funzip", 'args': ""},
+    '.tar': {'type': "tar",   'program': "tar",    'args': "-x -O"},
+}
+
+# These need to be checked *before* those in DECOMPRESSER_EXTENSIONS
+NESTED_DECOMPRESSER_EXTENSIONS = {
+    '.tar.bz2': {'program': "tar", 'args': "-x -j -O"},
+    '.tar.gz':  {'program': "tar", 'args': "-x -z -O"},
+    '.tar.xz':  {'program': "tar", 'args': "-x -J -O"},
+    '.tar.lzo': {'program': "tar", 'args': "-x --lzo -O"},
+    '.tar.lz4': {'program': "tar", 'args': "-x -Ilz4 -O"},
+}
```

Comment:
Can you either make it OrderedDict or use tuple of dictinaries or and ensure the order inside the data structure. For example:

ECOMPRESSER_EXTENSIONS = (
    {'.bz2': {'type': "bzip2", 'program': "bzip2",  'args': "-d -c"}},
    {'.gz':  {'type': "gzip",  'program': "gzip",   'args': "-d -c"}},
    ... and so on 
)